dsync/pull.go

```diff
@@ -99,6 +100,44 @@ func (f *Pull) Do(ctx context.Context) (err error) {
 }
 
 func (f *Pull) do(ctx context.Context) error {
+	protoID, err := f.remote.ProtocolVersion()
+	if err != nil {
+		return err
+	}
+
+	if protocolSupportsDagStreaming(protoID) {
+		if streamable, ok := f.remote.(DagStreamable); ok {
+			progCh := make(chan cid.Cid)
+			go func() {
+				for {
+					select {
+					case cid := <-progCh:
+						// this is the only place we should modify progress after creation
+						for i, hash := range f.info.Manifest.Nodes {
+							if cid.String() == hash {
+								f.prog[i] = 100
+								go f.completionChanged()
+								break
+							}
+						}
+					case <-ctx.Done():
+						return
+					}
+				}
+			}()
+
+			r, err := streamable.OpenBlockStream(ctx, f.info, f.meta)
+			if err != nil {
+				return err
+			}
+			defer r.Close()
+
+			_, err = AddAllFromCARReader(ctx, f.bapi, r, progCh)
+			return err
+		}
+		log.Debugf("protocol supports streaming but doesn't have the streamable interface: %T %v", f.remote, f.remote)
```

Comment:
Should this be more of an error condition/log?

Reply:
I've been avoiding using error-level logging in these sub packages because it really confuses users when it bleeds up into CLI land. Another solution would be to explicitly set the logging level up in qri land, but we should chat about that more. For now I'd like to keep these at the debug level

Reply:
Sounds good, we do have a bit of logging work ahead of us.